addons/mail/static/src/components/message/message.js

```diff
@@ -176,14 +176,14 @@ class Message extends Component {
      *
      * @returns {boolean}
      */
-    get hasDifferentOriginThread() {
+    get isOriginatingFromCurrentThread() {
         if (!this.threadViewer) {
-            return false;
+            return true;
         }
         if (!this.message.originThread) {
-            return false;
+            return true;
         }
-        return this.message.originThread !== this.threadViewer.thread;
+        return this.message.originThread === this.threadViewer.thread;
```

Comment:
Now that I see it written this is actually tricky.

I don't think message should be considered originating from current thread if there is no current thread or no known origin thread.

I feel like we really want 3 outcomes here.

Either we allow undefined as preemptive return and test for strictly true or strictly false, or we actually have 2 methods: one for the positive outcome, and one for the negative outcome, with the preemptive return being false in both cases.

cc @alexkuhn?

Reply:
Or we just move both conditions separately on the template:
`this.threadViewer && this.message.originThread && this.message.originThread === this.threadViewer.thread`
`this.threadViewer && this.message.originThread && this.message.originThread !== this.threadViewer.thread`

Reply:
Unpopular opinion, but I prefer not having such a getter and instead simply use `message.originThread` and `threadViewer.thread` in template altogether.

Yes, this more verbose and has "perceived duplicated" code, but this has the advantage of being explicit (including how to treat message without an origin thread, or rendering of messaging not in a context of a thread), and it doesn't add an indirection.